src/apps/recipes/pages/new/index.tsx

```diff
@@ -0,0 +1,85 @@
+import { useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { getUserState } from '../../../users/services/user_service';
+import * as S from './styles';
+import * as D from '../../components/recipe_view/styles';
+
+import { ActionButton, InputField } from '../../../../shared/components';
+import { UpdateRecipe, IngredientDTO } from '../../types/recipes';
+import { createRecipe } from '../../services/recipe_service';
+
+export const NewRecipe = () => {
+  const history = useHistory();
+
+  useEffect(() => {
+    // If the user is not logged in, redirect to the login page
+    // TODO: Send a notification to the user saying that they need to be logged in. Maybe as a banner?
```

Comment:
FYI: We use an in-house `Toast` component to display these types of messages to the user which is quite easy to use. I don't think it's worth your time replicating that here, but just so you know.